### PR TITLE
docs: Update README.md & add CONTRIBUTING.md

### LINKED ISSUE #1289: Improve README.md and CONTRIBUTING.md
For better experience for new contributors we should improve our "how to get started" documentation inside the repository. (Basically the files `README.md` and `CONTRIBUTING.md`.)

Cool would be that a new developer can find all the information she needs on how to set up her local environment, run tests and start to contribute.

It has been done already for [sentry-python](https://github.com/getsentry/sentry-python) so we can steal a lot of stuff from over there.

This is how those files should look like (If the section is not applicable to the SDK, we can ommit it.)

### README.md
- Sentry header (if not already there copy from sentry-python)
- badges of coverage etc for different versions/integrations (if available)]
- Headline1: Getting started
    - Headline2: Requirements (describe if there are any requirements that need to met in order for the SDK to work.)
    - Headline2: Install (how to install the SDK to the local dev environment)
    - Headline2: Configuration (how to add the SDK to your project)
    - Headline2: Usage (how to verify configuration, enriching stuff and so on.)
- Headline1: Integrations (list of all integrations (if any) with links to documentation)
- Headline1: Migrate from {old version name} (if there is a old raven version or some other incompatible old version)
- Headline1: Contributing to the SDK (link to [CONTRIBUTING.md](http://contributing.md/))
- Headline1: Getting Help/Support (describing how to get help/support. just copy text from sentry-python repository)
- Headline1: Resources (links to places on the web)
- Headline1: License (link to license file. see sentry-python repository)

(as a template on how this should look like see: [https://github.com/getsentry/sentry-python/blob/master/README.md](https://github.com/getsentry/sentry-python/blob/master/README.md))

### CONTRIBUTING.md
- intro text and link to how to contribute to documentation (copy from sentry-python repository)
- Headline1: How to report a problem (copy from sentry-python repository)
- Headline1: Submitting changes (copy from sentry-python repository)
- Headline1: Development Environment (how to set up, step by step guide. should be as easy as possible so that novice devs can do it. should work when copy pasting into shell on a fresh computer)
- Headline1: Running Tests (step by step guide. should work when copy pasting into shell on a fresh computer)
- Headline1: Releasing a new Version (copy from and ammend from sentry-python if this SDK also uses craft for deployment)
    - Headline2: Versioning Policy (copy from sentry-python, if this SDK also uses semver)
- Headline1: Commit message guideline (link to docs, copy from sentry-python)

(as a template on how this should look like see: [https://github.com/getsentry/sentry-python/blob/master/CONTRIBUTING.md](https://github.com/getsentry/sentry-python/blob/master/CONTRIBUTING.md))

### PR DESCRIPTION
**Pull request by cleptric**
This is based on our [Laravel SDK CONTRIBUTING.md](https://github.com/getsentry/sentry-laravel/blob/master/CONTRIBUTING.md).

Please let me know if I should add more things or if anything in here is plain wrong.

Closes #1289

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+### Clone the repository
+
+```
+$ git clone git@github.com:getsentry/sentry-php.git
+```
+
+Make sure that you have PHP 7.2 installed, as this is the lowest version the SDK should work with.
+
+On macOS, we recommend using [Homebrew](https://brew.sh/) to install PHP. For Windows, we recommend ???.
```
Comment (cleptric) [inline CONTRIBUTING.md:38]: How do people install PHP on Windows these days? Is XAMPP still a thing?

Comment (ste93cry) [reply CONTRIBUTING.md:38]: I think that either they use a tool like XAMPP or they download the pre-built PHP from https://windows.php.net or (most probably) they use Docker

**Comment by ste93cry** [comment]
I wonder if we should really enter the rabbit holes of how to install PHP, rather than just saying that it is a prerequisite for the working environment (of course)

**Comment by stayallive** [comment]
I agree, we shouldn't provide a step-by-step installation guide for PHP plenty of info on that on the internet, but we can give some guidance. Maybe some inspration can be taken from the Laravel SDK where I updated the guide few months ago

- https://github.com/getsentry/sentry-laravel/blob/master/CONTRIBUTING.md
- https://github.com/getsentry/sentry-laravel/pull/548

**Comment by cleptric** [comment]
I aligned the `CONTRIBUTING.md` with our Laravel SDK.
I added our branch workflow as well as a note about updating the `CHANGELOG.md` file.
Once this is merged, let's also update `sentry-laravel` and `sentry-symfony`.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+## How to report a problem
+
+Please search the [issue tracker](https://github.com/getsentry/sentry-php/issues) before creating a new issue (a problem or an improvement request). Please also ask in our [Sentry Community on Discord](https://discord.com/invite/Ww9hbqr) before submitting a new issue. There is a ton of great people in our Discord community ready to help you!
+
+If you feel that you can fix or implement it yourself, please read a few paragraphs below to learn how to submit your changes.
+
+## Submitting changes
+
+- Setup the development environment.
+- Clone the sentry-php repository and prepare necessary changes.
```
Comment (stayallive) [inline CONTRIBUTING.md:20]:
```suggestion
- Clone the `sentry-php` repository and prepare necessary changes.
```

**Review by vladanpaunovic — APPROVED** [review]
I agree with above sentiments from @stayallive and @ste93cry (thanks folks!).

Other than that, this looks good. Nice addition to this repo! Thanks @cleptric! 🥳

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+# Contributing to the Sentry PHP SDK
+
+We welcome contributions to sentry-php by the community. See the [Contributing to Docs](https://docs.sentry.io/contributing/) page if you want to fix or update the documentation on the website.
+
+## How to report a problem
+
+Please search the [issue tracker](https://github.com/getsentry/sentry-php/issues) before creating a new issue (a problem or an improvement request). Please also ask in our [Sentry Community on Discord](https://discord.com/invite/Ww9hbqr) before submitting a new issue. There is a ton of great people in our Discord community ready to help you!
+
+If you feel that you can fix or implement it yourself, please read a few paragraphs below to learn how to submit your changes.
```
[the remark this answers is not in the document]
Comment (cleptric) [reply CONTRIBUTING.md:15]: Looking at https://github.com/getsentry/sentry-python/blob/master/CONTRIBUTING.md, I'm not sure if we should try to align our SDKs or if it's ok to deviate slightly. @vladanpaunovic How do we normally handle these cases?

[the remark this answers is not in the document]
Comment (stayallive) [reply CONTRIBUTING.md:15]: I mean I agree, but maybe the Python one is wrong too.

Seems a bit odd to link to the development docs of how to change the documentation site from the first paragraph in the SDK contributing docs right?

[the remark this answers is not in the document]
Comment (cleptric) [reply CONTRIBUTING.md:15]: I do agree 😄

**Comment by cleptric** [comment]
Based on #1289, I made some changes to the README.md as well.

**Review by stayallive — APPROVED** [review]
Looks good, we can always keep iterating on this 😄

**Pull request merged**